src/components/progress-steps/progress-steps.jsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import { cn } from '../../utilities/functions';
+import { Check } from 'lucide-react';
+
+/**
+ * ProgressSteps Component
+ */
+const ProgressSteps = ({
+    variant = 'dot', // dot, number, icon
+    size = 'sm',     // sm, md, lg
+    type = 'inline', // inline, stack
+    currentStep = 1, 
+    variantIcon,   
+    children,
+    className,
+    ...props
+}) => {
+    const totalSteps = React.Children.count(children);
+
+    const sizeClasses = {
+        sm: {
+            dot: 'w-2.5 h-2.5',
+            ring: 'w-5 h-5',
+            numberIcon: 'w-5 h-5 text-[10px]',
+            line: type === 'stack' ? 'h-0.5' : 'h-0.5 mx-2', 
+            icon: 'w-5 h-5',
+        },
+        md: {
+            dot: 'w-3 h-3',
+            ring: 'w-6 h-6',
+            numberIcon: 'w-6 h-6 text-[12px]',
+            line: type === 'stack' ? 'h-0.5' : 'h-0.5 mx-3', 
+            icon: 'w-6 h-6',
+        },
+        lg: {
+            dot: 'w-3.5 h-3.5',
+            ring: 'w-7 h-7',
+            numberIcon: 'w-7 h-7 text-[14px]',
+            line: type === 'stack' ? 'h-0.5' : 'h-0.5 mx-2', 
+            icon: 'w-7 h-7',
+        },
+    };
+
+    const steps = React.Children.toArray(children).map((child, index) => {
+        const isCompleted = index + 1 < currentStep;
+        const isCurrent = index + 1 === currentStep;
+
+        const stepContent = variant === 'dot' ? (
```

Comment:
Avoid using nested ternary operators, use functions instead for better code readability.

Reply:
ok